src/main/scala/org/apache/spark/sql/delta/analysis/DeltaAnalysis.scala

```diff
@@ -32,11 +33,16 @@ import org.apache.spark.sql.util.SchemaUtils
 
 case object DeltaAnalysis extends Rule[LogicalPlan] {
 
-  private def needsSchemaAdjustment(query: LogicalPlan, schema: StructType): Boolean = {
+  private def needsSchemaAdjustment(
```

Comment:
nit: for future maintainability we should have a comment about when and why schema adjustment is needed